services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/security/AdGroupValidatorTest.kt

```diff
@@ -58,6 +62,47 @@ class AdGroupValidatorTest {
         assertThat(result).isSuccess()
     }
 
+    @ParameterizedTest
+    @MethodSource("emptyNames")
+    fun `fails when AD token has no given name information`(
+        useAdFilter: Boolean,
+        givenName: String?,
+    ) {
+        val validator = validator(useAdFilter)
+        val jwt =
+            AuthenticationMocks.adJwt(
+                adGroups = AuthenticationMocks.DEFAULT_AD_GROUPS,
+                givenName = givenName,
+            )
+
+        val result = validator.validate(jwt)
+
+        assertThat(result).hasError("Missing given_name")
+    }
+
+    @ParameterizedTest
+    @MethodSource("emptyNames")
+    fun `fails when AD token has no family name information`(
+        useAdFilter: Boolean,
+        familyName: String?,
+    ) {
+        val validator = validator(true)
```

Comment:
Should this be `val validator = validator(useAdFilter)`?

Reply:
Yes, it should. Thank you for noticing. Fixed now.